messaging/src/test/java/org/axonframework/commandhandling/gateway/AbstractRetrySchedulerTest.java

```diff
@@ -0,0 +1,116 @@
+package org.axonframework.commandhandling.gateway;
+
+import org.axonframework.commandhandling.CommandExecutionException;
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.common.AxonNonTransientException;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Predicate;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class AbstractRetrySchedulerTest {
+
+  public static class RetrySchedulerStub extends AbstractRetryScheduler {
+
+    protected RetrySchedulerStub(Builder builder) {
+      super(builder);
+    }
+
+    @SuppressWarnings("rawtypes")
+    @Override
+    protected long computeRetryInterval(CommandMessage commandMessage, RuntimeException lastFailure, List<Class<? extends Throwable>[]> failures) {
+      return 100;
+    }
+
+    static Builder builder() {
+      return new Builder();
+    }
+
+    public static class Builder extends AbstractRetryScheduler.Builder<Builder> {
+      RetrySchedulerStub build() {
+        return new RetrySchedulerStub(this);
+      }
+    }
+  }
+
+  @Test
+  void isExplicitlyNonTransient_defaults() {
+    RetrySchedulerStub retrySchedulerStub = RetrySchedulerStub.builder().retryExecutor(mock(ScheduledExecutorService.class)).build();
+    assertTrue(
+        retrySchedulerStub.isExplicitlyNonTransient(new AxonNonTransientException("message") {}),
+        "AxonNonTransientException should be treated as non-transient by default"
+    );
+  }
+
+  @Test
+  void isExplicitlyNonTransient_configuredNonTransientFailures() {
+    RetrySchedulerStub retrySchedulerStub = RetrySchedulerStub
+        .builder()
+        .retryExecutor(mock(ScheduledExecutorService.class))
+        .nonTransientFailures(Arrays.asList(AxonNonTransientException.class, CommandExecutionException.class))
+        .build();
+
+    assertTrue(
+        retrySchedulerStub.isExplicitlyNonTransient(new AxonNonTransientException("message") {}),
+        "Per configuration, AxonNonTransientException should be treated as non-transient"
+    );
+
+    assertTrue(
+        retrySchedulerStub.isExplicitlyNonTransient(new CommandExecutionException("message", null)),
+        "Per configuration, CommandExecutionException should be treated as non-transient"
+    );
+  }
+
+  @Test
+  void isExplicitlyNonTransient_configuredNonTransientFailurePredicate() {
```

Comment:
What about adding a test that validates the list is ignored when the predicate is present? This test does that to some extent, but it could be a little more upfront by actually adding the list too.